homeassistant/components/bosch_shc/sensor.py

```diff
@@ -0,0 +1,401 @@
+
+
+class TemperatureSensor(SHCEntity):
+    """Representation of a SHC temperature reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_temperature"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Temperature"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return TEMP_CELSIUS
+
+
+class HumiditySensor(SHCEntity):
+    """Representation of a SHC humidity reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_humidity"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Humidity"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.humidity
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return PERCENTAGE
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:water-percent"
+
+
+class PuritySensor(SHCEntity):
+    """Representation of a SHC purity reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_purity"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Purity"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.purity
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return CONCENTRATION_PARTS_PER_MILLION
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:molecule-co2"
+
+
+class AirQualitySensor(SHCEntity):
+    """Representation of a SHC airquality reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_airquality"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Air Quality"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.combined_rating.name
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "rating_description": self._device.description,
+            "temperature_rating": self._device.temperature_rating.name,
```

Comment:
Is this temperature measurement not available as a separate sensor for this device?

Reply:
The ratings are an enum with possible values `Good`, `Medium`, `Bad`. For the actual temperature, a temperature entity is already added above.

Reply:
Ok. I think we should make these separate sensors unless they are dependent on the main sensor state measurement somehow.